how-to/synonym-expansion.md

```diff
@@ -0,0 +1,150 @@
+
+### How to make a new table to register synonyms
+
+We make a table to register synonyms as below.
+
+```sql
+CREATE TABLE synonyms (
+  term text PRIMARY KEY,
+  synonyms text[]
+);
+
+CREATE INDEX synonyms_search ON synonyms USING pgroonga (term pgroonga.text_term_search_ops_v2);
+```
+
+We can get a term that is registered in `synonyms` with a key that is registered in `term`. 
+
+For example, if we want to also match "display" when specifying "window" as the search keyword, register "window" to `term` and register "window" and "display" to `synonyms`.
+(We attention to need also register target synonym term to `synonyms`.)
+
+We set PGroonga's index to `term` for searching to fast and without uppercase and lowercase.
+
+We can register multiple synonyms into `synonyms` because type of `synonyms` is `text[]`.
+
+### Register synonyms
+
+We can register synonyms by inserting data into `synonyms`.
+
+For example, if we want to register "Window", "display", and "video display" as synonyms, we can insert these terms into the table that register synonyms as below.
+
+```sql
+INSERT INTO synonyms (term, synonyms) VALUES ('Window', ARRAY['Window', 'display', 'video display']);
+INSERT INTO synonyms (term, synonyms) VALUES ('display', ARRAY['display', 'Window', 'video display']);
+INSERT INTO synonyms (term, synonyms) VALUES ('video display', ARRAY['video display', 'Window', 'display']);
+```
+
+### Add synonyms
+
+There are three patterns to add synonyms.
+
+#### Add new synonyms
+
+We can add synonyms in the same way as register synonyms.
+For example, if we want to add "copy", "replicate", and "simulate" as synonyms, we insert these terms into the table that register synonyms as below.
+
+```sql
+INSERT INTO synonyms (term, synonyms) VALUES ('copy', ARRAY['copy', 'replicate', 'simulate']);
+INSERT INTO synonyms (term, synonyms) VALUES ('replicate', ARRAY['replicate', 'copy', 'simulate']);
+INSERT INTO synonyms (term, synonyms) VALUES ('simulate', ARRAY['simulate', 'copy', 'replicate']);
+```
+
+#### Add new synonyms in existing one
+
+If we want to add new synonyms in the existing one, we update the existing record.
+For example, if we also want to match "imitate" when we use "copy" as a search key word.
+
+```sql
+UPDATE synonyms SET synonyms = array_append(synonyms, 'imitate') WHERE term = 'copy';
+UPDATE synonyms SET synonyms = array_append(synonyms, 'imitate') WHERE term = 'replicate';
+UPDATE synonyms SET synonyms = array_append(synonyms, 'imitate') WHERE term = 'simulate';
+```
+
+"imitate" is new synonym. Therefore we add record into `synonyms` by using `INSERT` as below.
+
+```sql
+INSERT INTO synonyms (term, synonyms) VALUES ('imitate', ARRAY['imitate', 'copy', 'replicate', 'simulate']);
+```
+
+#### Modify exist synonyms
+
+If we want to modify an existing record, we update one.
+For example, if we want to modify "Windows" to "Window", we as below.
```

Comment:
For example, if we want to modify "Windows" to "Window", we **do** as below.

_Do_ or _something verb_ is required between **we** and **as below**.